module/VuFind/tests/unit-tests/src/VuFindTest/Controller/InstallControllerTest.php

```diff
@@ -0,0 +1,198 @@
+     */
+    public function testGetMinimalPhpVersionWithMissingFile()
+    {
+        $controller = $this->mockController([]);
+        $method = $this->getMinimalPhpVersionMethod();
+        $this->expectException(\Exception::class);
+        $this->expectExceptionMessage('Cannot find composer.json');
+        $method->invokeArgs($controller, []);
+    }
+
+    /**
+     * Simulate no PHP version defined in composer.json file
+     *
+     * @return void
+     */
+    public function testGetMinimalPhpVersionWithMissingPhpVersion()
+    {
+        $controller = $this->mockController(['name' => 'vufind/vufind']);
+        $method = $this->getMinimalPhpVersionMethod();
+        $this->expectException(\Exception::class);
+        $this->expectExceptionMessage('Cannot parse PHP version from composer.json');
+        $method->invokeArgs($controller, []);
+    }
+
+    /**
+     * getMinimalPhpVersion test data
+     *
+     * @return array[]
+     */
+    public static function getMinimalPhpVersionProvider(): array
+    {
+        return [
+            [
+                [
+                    'require' => [
+                        'php' => '>=7.4.1',
+                    ],
+                ],
+                '7.4.1',
+            ],
+            [
+                [
+                    'require' => [
+                        'php' => '7.3.0',
+                    ],
+                ],
+                '7.3.0',
+            ],
+            [
+                [
+                    'require' => [
+                        'php' => '^7.2.0',
+                    ],
+                ],
+                '7.2.0',
+            ],
+            [
+                [
+                    'require' => [
+                        'php' => '~7.1.0',
+                    ],
+                    'config' => [
+                        'platform' => [
+                            'php' => '5.6.0',
+                        ],
+                    ],
+                ],
+                '7.1.0',
+            ],
+            [
+                [
+                    'config' => [
+                        'platform' => [
+                            'php' => '7.0.0',
+                        ],
+                    ],
+                ],
+                '7.0.0',
+            ],
+        ];
+    }
+
+    /**
+     * Test getMinimalPhpVersion with actual composer.json file
+     *
+     * @dataProvider getMinimalPhpVersionProvider
+     * @return void
+     */
+    public function testGetMinimalPhpVersion($json, $expected)
+    {
+        $controller = $this->mockController($json);
+        $method = $this->getMinimalPhpVersionMethod();
+        $this->assertEquals(
+            $expected,
+            $method->invokeArgs($controller, [])
+        );
+    }
+
+    /**
+     * Mock controller
+     *
+     * @param array $json JSON data
+     *
+     * @return InstallController
+     */
+    protected function mockController(array $json): InstallController
```

Comment:
I'd rename this to getMockControllerWithComposerJson or something else more specific, since this method is only really relevant to tests that are working with composer.json, and if we expand the test suite, we may want to create different mock controllers for different purposes.